src/core/src/core_logic/PatchInstaller.py

```diff
@@ -801,3 +803,31 @@ def get_max_batch_size(self, maintenance_window, package_manager):
 
         return max_batch_size_for_packages
 
+    def __log_progress_status(self, maintenance_window, installed_update_count):
+        progress_status = self.progress_template.format(str(datetime.timedelta(minutes=maintenance_window.get_remaining_time_in_minutes())), str(self.attempted_parent_package_install_count), str(self.successful_parent_package_install_count), str(self.failed_parent_package_install_count), str(installed_update_count - self.successful_parent_package_install_count),
+                                                        "Completed processing packages!")
+        self.composite_logger.log(progress_status)
+        
+    def __send_metadata_to_health_store(self):
+        """ store patch metadata in status files for health store. """
+        self.composite_logger.log_debug("[PI] Reviewing final healthstore record write. [HealthStoreId={0}][MaintenanceRunId={1}]".format(str(self.execution_config.health_store_id), str(self.execution_config.maintenance_run_id)))
+        if self.execution_config.health_store_id is not None:
+            self.status_handler.set_patch_metadata_for_healthstore_substatus_json(
+                patch_version=self.execution_config.health_store_id,
+                report_to_healthstore=True,
+                wait_after_update=False)
+
+    # region - Failed packages retry succeeded
+    def __log_final_warning_metric(self):
+        """ Log the final metrics for status handler error json for setting installation status to warning. """
+
+        message = "All requested package(s) are installed. Any patch errors marked are from previous attempts."
+        self.composite_logger.log(message)
+        self.status_handler.add_error_to_status(message=message, error_code=Constants.PatchOperationErrorCodes.PACKAGES_RETRY_SUCCEEDED)
+    
+    def should_patch_installation_status_be_set_to_warning(self, patch_installation_successful, maintenance_window_exceeded):
+        """ Check if patch installation status can be set to warning from failed. """
+        # type (bool, bool) -> bool
+        return not patch_installation_successful and not maintenance_window_exceeded and self.status_handler.are_all_requested_packages_installed()
```

Comment:
@kjohn-msft I think we also need to check if reboot is pending here. We have an existing function mark_installation_completed() which also sets installation status to warning based on a few conditions, wondering if we should merge this one with that
<img width="749" alt="image" src="https://github.com/user-attachments/assets/837dee22-8e55-4e29-b1a0-4b9883a70cb8" />